src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColLevelTwoService.java

```diff
@@ -0,0 +1,82 @@
+package uk.ac.ebi.eva.evaseqcol.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import uk.ac.ebi.eva.evaseqcol.entities.SeqColExtendedDataEntity;
+import uk.ac.ebi.eva.evaseqcol.entities.SeqColLevelOneEntity;
+import uk.ac.ebi.eva.evaseqcol.entities.SeqColLevelTwoEntity;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class SeqColLevelTwoService {
+
+    @Autowired
+    private SeqColExtendedDataService extendedDataService;
+
+    @Autowired
+    private SeqColLevelOneService levelOneService;
+
+    /**
+     * Make 2 recursive lookups to retrieve and construct the seqCol level 2 object
+     * @param digest: level 0 seqCol digest*/
+    public Optional<SeqColLevelTwoEntity> getSeqColLevelTwoByDigest(String digest) {
+        // 1 DATABASE LOOKUP
+        Optional<SeqColLevelOneEntity> levelOneEntity = levelOneService.getSeqColLevelOneByDigest(digest);
+        if (!levelOneEntity.isPresent()) {
+            //TODO THROW EXCEPTION
+            System.out.println("EXCPETION: seqCol with digest: " + digest + "doesn't exists !");
```

Comment:
**Optional**: Use standard loggers throughout the code base...

Reply:
True..I'll do that.